_docs/reference/config/kubernetes-proxy-injection.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Kubernetes sidecar proxy injection
+overview: Kubernetes sidecar proxy injection
+
+order: 15
+
+layout: docs
+type: markdown
+---
+
+Each pod in the Kubernetes cluster must be running an Istio compatible 
+sidecar. The following sections describe two ways of injecting the Istio 
+sidecar into a pod: manually using istioctl `kube-inject` or automatically 
+using the Istio sidecar injector webhook. 
+
+Manual injection modifies the controller configuration, e.g. deployment. It 
+does this by modifying the pod template spec such that *all* pods for that
+deployment are created with the injected sidecar. Adding/Updating/Removing
+the sidecar requires modifying the entire deployment.
+
+Automatic injection injects at pod creation time. The controller resource is 
+unmodified. Sidecars can be selectively added/updated/removed by (re)starting 
+pods after modifying the injection configuration.
+
+Manual and automatic injection use the same templated configuration. See TBD 
+for template configuration format.
+
+# Manual injection
+
+`kube-inject` is designed to be run offline without access to a running Kubernetes
+cluster. 
```

Comment:
This is not what 0.4 is:  
```$ ./istioctl kube-inject -f ../samples/bookinfo/kube/bookinfo.yaml
ERROR: logging before flag.Parse: I0126 12:50:38.167787   30394 client.go:48] using in-cluster configuration
ERROR: logging before flag.Parse: W0126 12:50:38.169168   30394 client_config.go:529] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
ERROR: logging before flag.Parse: W0126 12:50:38.169407   30394 client_config.go:534] error creating inClusterConfig, falling back to default config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined. ```

Reply:
Right. I tried to fix the implicit/default runtime dependency on k8s api-server in 0.5.0. The k8s client and ConfigMap access should only occur if `--meshConfigMapName` is explicitly specified.

Reply:
ok cool, that is a great move!